docs/dev/clusteroperator.md

```diff
@@ -85,7 +91,7 @@ If the controller reaches 4.0.1, the conditions might be:
 If an error blocks reaching 4.0.1, the conditions might be:
 
 * `Failing` is true with a detailed message `Unable to apply 4.0.1: could not update 0000_70_network_deployment.yaml because the resource type NetworkConfig has not been installed on the server.`
-* `Available` is false with message `Cluster is unable to reach 4.0.1`
-* `Progressing` is true with message `Unable to apply 4.0.1: a required object is missing`
```

Comment:
this one changed because it logically required an error flow feeding into a progressing message decision which is duplicating information.

Reply:
This is as designed, and it's not duplication, it's summarization.

Progressing tells you what is happening.  Available tells you why it's unavailable.  Failing tells you the details.

Reply:
my interpretation/understanding was:

available tells you if the operand is available or why not if not

progressing tells you if the operand is being modified/reconfigured (and if so, why)

failing tells you the operator has given up on managing/configuring/reconfiguring the operand for some reason (and why).

ie as @deads2k said, they are orthogonal dimensions, not (necessarily) correlated information/explanations.